### PR TITLE
Rule Dev

### PR DESCRIPTION
**Pull request by nasbench**
This PR covers the following:

- General updates to some rules such as: metadata, logic, missing quotes, missing `OriginalFileName` field
- Updated the `ftp.exe` rule
  - Renamed the rule from `susp`prefix to the `lolbin`
  - Fixed the logic of the `endswith`modifier
  - Separated the `renamed` execution version to another new rule
- Added new rule to cover the usage of `sftp.exe` as a LOLBIN
- Added new rules related to Code Integrity events
- Other quality of life updates :) (See files changed section for full details)

### CODE REVIEW

**Review by frack113 — APPROVED** [review]

File: rules/windows/builtin/code_integrity/win_codeintegrity_attempted_dll_load.yml [hunk 1]
```diff
@@ -1,10 +1,11 @@
 title: Code Integrity Attempted DLL Load
 id: f8931561-97f5-4c46-907f-0a4a592e47a7
-description: Detects attempted DLL load events that didn't meet signing level requirements
+description: Detects attempted DLL load events that didn't meet Antimalware or Windows signing level requirements. It often means the file's signature is revoked or expired
```
Comment (phantinuss) [inline rules/windows/builtin/code_integrity/win_codeintegrity_attempted_dll_load.yml:3]:
```suggestion
description: Detects attempted DLL load events that didn't meet anti-malware or Windows signing level requirements. It often means the file's signature is revoked or expired
```

File: rules/windows/builtin/code_integrity/win_codeintegrity_revoked_driver.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+logsource:
+    product: windows
+    service: codeintegrity-operational
+detection:
+    selection:
+        EventID: 3023
+    condition: selection
+falsepositives:
+    - Unknown
+level: critical
```
Comment (phantinuss) [inline rules/windows/builtin/code_integrity/win_codeintegrity_revoked_driver.yml:21]: I am not sure about critical. High feels more suitable as a revoked driver load can only just mean old/shitty software (or am I wrong?) which would put it to high in my books
```suggestion
level: high
```

File: rules/windows/builtin/code_integrity/win_codeintergiry_blocked_driver_load.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+title: Code Integrity Blocked Driver Load
+id: e4be5675-4a53-426a-8c81-a8bb2387e947
+description: Detects blocked load events that did not meet the authenticode signing level requirements or violated code integrity ploicy
```
Comment (phantinuss) [inline rules/windows/builtin/code_integrity/win_codeintergiry_blocked_driver_load.yml:3]:
```suggestion
description: Detects blocked load events that did not meet the authenticode signing level requirements or violated code integrity policy
```

File: rules/windows/process_creation/proc_creation_win_computer_discovery_get_adcomputer.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+            - 'PowerShell.EXE'
+            - 'pwsh.dll'
+    selection_cli:
+        CommandLine|contains|all:
+            - 'Get-ADComputer '
+            - ' -Filter \*'
+    selection_output:
+        CommandLine|contains:
+            - ' > '
+            - 'Out-File'
```
Comment (phantinuss) [inline rules/windows/process_creation/proc_creation_win_computer_discovery_get_adcomputer.yml:31]:
```suggestion
        CommandLine|contains:
            - ' > '
            - 'Out-File'
            - 'Set-Content'
            - 'Add-Content'
```

File: rules/windows/process_creation/proc_creation_win_lolbin_sftp.yml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+detection:
+    selection:
+        Image|endswith: '\sftp.exe' # The "sftp.exe" located in the OpenSSH directory has no OriginalFileName :(
+        CommandLine|contains:
+            # Since "-D" is a valid flag for other usage we assume the user is going to enter a path
+            # Either a full one like "C:\Windows\System32\calc.exe" or a relative one "..\..\..\Windows\System32\calc.exe"
+            # In my testing you can't execute direct binaries by their name via this method (if you found a way please update the rule)
+            - ' -D ..'
+            - ' -D C:\'
+    condition: all of selection*
```
Comment (phantinuss) [inline rules/windows/process_creation/proc_creation_win_lolbin_sftp.yml:25]:
```suggestion
    condition: selection
```

File: rules/windows/process_creation/proc_creation_win_renamed_ftp.yml [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    - attack.t1202
+logsource:
+    category: process_creation
+    product: windows
+detection:
+    selection_img:
+        Image|endswith: '\ftp.exe'
+    selection_original:
+        OriginalFileName: 'ftp.exe'
+    condition: selection_original and not selection_img
```
Comment (phantinuss) [inline rules/windows/process_creation/proc_creation_win_renamed_ftp.yml:23]:
```suggestion
    selection_original:
        OriginalFileName: 'ftp.exe'
    filter_img:
        Image|endswith: '\ftp.exe'
    condition: selection_original and not filter_img
```

File: rules/windows/process_creation/proc_creation_win_sc_query.yml [hunk 7]
```diff
@@ -0,0 +1,25 @@
+title: Sc Query Execution
```
Comment (phantinuss) [inline rules/windows/process_creation/proc_creation_win_sc_query.yml:1]:
```suggestion
title: SC.EXE Query Execution
```

File: rules/windows/process_creation/proc_creation_win_susp_copy_lateral_movement.yml [hunk 8]
```diff
@@ -20,15 +21,23 @@ logsource:
...
+        - Image|contains:
+            - '\powershell.exe'
+            - '\pwsh.exe'
+        - OriginalFileName:
+            - 'PowerShell.EXE'
+            - 'pwsh.dll'
+    selection_pwsh_cli:
         CommandLine|contains:
             - 'copy-item'
             - 'copy'
```
Comment (phantinuss) [inline rules/windows/process_creation/proc_creation_win_susp_copy_lateral_movement.yml:43]: I cannot make github to edit "unshown" areas. Please add:
```
'move-item'
' mi '

```

File: rules/windows/process_creation/proc_creation_win_user_discovery_get_aduser.yml [hunk 9]
```diff
@@ -24,6 +26,10 @@ detection:
         CommandLine|contains|all:
             - 'Get-ADUser '
             - ' -Filter \*'
+    selection_output:
+        CommandLine|contains:
+            - ' > '
+            - 'Out-File'
```
Comment (phantinuss) [inline rules/windows/process_creation/proc_creation_win_user_discovery_get_aduser.yml:32]:
```suggestion
        CommandLine|contains:
            - ' > '
            - 'Out-File'
            - 'Set-Content'
            - 'Add-Content'
```

**Review by phantinuss — CHANGES_REQUESTED** [review]

File: rules/windows/process_creation/proc_creation_win_susp_copy_lateral_movement.yml (hunk 8 above)
Comment (nasbench) [reply rules/windows/process_creation/proc_creation_win_susp_copy_lateral_movement.yml:43]: Done

**Review by phantinuss — APPROVED** [review]

**Pull request merged**